src/MahApps.Metro/MahApps.Metro.Shared/Controls/Helper/TextBoxHelper.cs

```diff
@@ -247,24 +252,54 @@ private static void UseSpellCheckContextMenuChanged(DependencyObject d, Dependen
                 // set the spell check to true
                 tb.SetValue(SpellCheck.IsEnabledProperty, true);
                 // override pre defined context menu
-                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu();
+                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu(tb);
                 tb.ContextMenuOpening += TextBoxBaseContextMenuOpening;
             }
             else
             {
                 tb.SetValue(SpellCheck.IsEnabledProperty, false);
-                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu();
+                tb.ContextMenu = GetDefaultTextBoxBaseContextMenu(tb);
                 tb.ContextMenuOpening -= TextBoxBaseContextMenuOpening;
             }
         }
 
+        private static void IsCutCopyPasteInContextMenuChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var tb = d as TextBoxBase;
+            if (null == tb)
+            {
+                throw new InvalidOperationException("The property 'IsCutCopyPasteInContextMenuChanged' may only be set on TextBoxBase elements.");
+            }
+
+            tb.SetValue(IsCutCopyPasteInContextMenuProperty, e.NewValue);
```

Comment:
Is this really necessary? I'd think this method is called **because** the `IsCutCopyPasteInContextMenu` was changed, so setting it again is not needed. But I may be wrong.